test/unit/messaging/messaging.spec.ts

```diff
@@ -3534,42 +3470,56 @@ describe('Messaging', () => {
       },
     ];
 
+    const GAPIC_RESPONSE = [
+      // An example IMessage that is normally returned by the generated client
+      // when sendMessage() is ran successfully
+      { name: 'projects/projec_id/messages/message_id' }
+    ];
+
     validMessages.forEach((config) => {
       it(`should serialize well-formed Message: ${config.label}`, () => {
-        // Wait for the initial getToken() call to complete before stubbing https.request.
-        return mockApp.INTERNAL.getToken()
-          .then(() => {
-            const resp = utils.responseFrom({ message: 'test' });
-            httpsRequestStub = sinon.stub(HttpClient.prototype, 'send').resolves(resp);
-            const req = config.req;
-            req.token = 'mock-token';
-            return messaging.send(req);
-          })
-          .then(() => {
-            const expectedReq = config.expectedReq || config.req;
-            expectedReq.token = 'mock-token';
-            expect(httpsRequestStub).to.have.been.calledOnce.and.calledWith({
-              method: 'POST',
-              data: { message: expectedReq },
-              timeout: 10000,
-              url: 'https://fcm.googleapis.com/v1/projects/project_id/messages:send',
-              headers: expectedHeaders,
-            });
-          });
+        generatedClientStub = sinon.stub(FcmServiceClient.prototype, 'sendMessage').resolves(GAPIC_RESPONSE);
+        const req = config.req;
+        req.token = 'mock-token';
+
+        return messaging.send(req).then(() => {
+          const expectedReq = config.expectedReq || config.req;
+          expectedReq.token = 'mock-token';
+          expectedHeaders;
+          expect(generatedClientStub).to.have.been.calledOnce.and.calledWith(
+            {
+              parent: 'projects/project_id',
+              message: {
+                data: expectedReq.data,
+                notification: expectedReq.notification,
+                fcmOptions: expectedReq.fcmOptions,
+                token: expectedReq.token,
+                //TODO: Expand for android, webpush, apns once functionality is added
+              },
+              validateOnly: undefined,
+            }
+          );
+        });
       });
     });
 
     it('should not throw when the message is addressed to the prefixed topic name', () => {
       return mockApp.INTERNAL.getToken()
         .then(() => {
-          const resp = utils.responseFrom({ message: 'test' });
-          httpsRequestStub = sinon.stub(HttpClient.prototype, 'send').resolves(resp);
+          //const resp = utils.responseFrom({ message: 'test' });
```

Comment:
Please remove.

Also the call to `INTERNAL.getToken()` should no longer be required.

Reply:
Removed!